### PR TITLE
feat: Support header and multiple headers

### PR DESCRIPTION
**Pull request by telenorpronstad**

### CODE REVIEW

File: src/main/kotlin/no/telenor/kt/restclient/util/util.kt [hunk 1]
```diff
@@ -13,10 +14,16 @@ import java.net.URI
...
+fun RequestBodySpec.json(value: Any) = this
 	.body(value)
 	.contentType(MediaType.APPLICATION_JSON)!!
 
+fun RequestBodySpec.header(header: String, value: Any) : RequestBodySpec = this
+	.header(header, value.toString())
+
+fun RequestBodySpec.headers(vararg headersWithValue: Pair<String, Any>) : RequestBodySpec = this
+	.headers { httpHeaders -> headersWithValue.forEach { httpHeaders.add(it.first, it.second.toString()) }}
+
```
Comment (jamesbradlee) [inline src/main/kotlin/no/telenor/kt/restclient/util/util.kt]: These header methods should extend `RequestHeaderSpec` and not `RequestBodySpec`. `RequestBodySpec` is used on POST, PATCH, DELETE, etc methods. `RequestHeaderSpec` is returned on GET, OPTIONS and HEAD methods. `RequestBodySpec` extends `RequestHeaderSpec`, so the headers will also become available on `RequestBodySpec` :)

File: src/main/kotlin/no/telenor/kt/restclient/util/util.kt [hunk 2]
```diff
@@ -13,10 +14,16 @@ import java.net.URI
...
 @Suppress("UNNECESSARY_NOT_NULL_ASSERTION")
-fun RestClient.RequestBodySpec.json(value: Any) = this
+fun RequestBodySpec.json(value: Any) = this
 	.body(value)
 	.contentType(MediaType.APPLICATION_JSON)!!
 
+fun RequestBodySpec.header(header: String, value: Any) : RequestBodySpec = this
+	.header(header, value.toString())
+
+fun RequestBodySpec.headers(vararg headersWithValue: Pair<String, Any>) : RequestBodySpec = this
```
Comment (jamesbradlee) [inline src/main/kotlin/no/telenor/kt/restclient/util/util.kt]: nit: the function could also be named just `header`, this way you don't have to remember to use a plural if you want to assign multiple.

**Review by jamesbradlee — CHANGES_REQUESTED** [review]
See conversations

File: src/main/kotlin/no/telenor/kt/restclient/util/util.kt (hunk 2 above)
Comment (telenorpronstad) [reply src/main/kotlin/no/telenor/kt/restclient/util/util.kt]: Well, I like to have the different names given how the Restclient interface is defined originally. Keeps the familiarity intact.

File: src/main/kotlin/no/telenor/kt/restclient/util/util.kt (hunk 1 above)
Comment (telenorpronstad) [reply src/main/kotlin/no/telenor/kt/restclient/util/util.kt]: oops :)

**Review by jamesbradlee — APPROVED** [review]
LGTM :+1: Well done!

**Pull request merged**